include/envoy/network/socket_interface.h

```diff
@@ -49,13 +49,17 @@ class SocketInterface {
 using SocketInterfacePtr = std::unique_ptr<SocketInterface>;
 
 /**
- * Create IoHandle for given address
+ * Create IoHandle for given address.
  * @param type type of socket to be requested
  * @param addr address that is gleaned for address type, version and socket interface name
  * @return @ref Network::IoHandlePtr that wraps the underlying socket file descriptor
  */
 static inline IoHandlePtr ioHandleForAddr(Socket::Type type,
                                           const Address::InstanceConstSharedPtr addr) {
+  // Cannot create IoHandle for internal address yet.
+  if (addr->envoyInternalAddress()) {
+    return nullptr;
```

Comment:
if this stands, there should be a unit test please.

Reply:
You are so right! My brain stop working last night so I pushed before I save this change.

The latest update has an integration test connecting to a internal address